src/docfx/build/moniker/MonikerProvider.cs

```diff
@@ -81,36 +83,49 @@ public SourceInfo<string> GetConfigMonikerRange(FilePath file)
             return (null, monikers);
         }
 
-        private (Error? error, string[] monikers) GetFileLevelMonikersCore(FilePath file)
+        private (List<Error> errors, string[] monikers) GetFileLevelMonikersCore(FilePath file)
         {
+            var errors = new List<Error>();
             var (_, metadata) = _metadataProvider.GetMetadata(file);
 
             var configMonikerRange = GetConfigMonikerRange(file);
-            var configMonikers = _rangeParser.Parse(configMonikerRange);
+            var configedMonikers = _rangeParser.Parse(configMonikerRange);
 
-            if (!string.IsNullOrEmpty(metadata.MonikerRange))
+            if (!string.IsNullOrEmpty(metadata.MonikerRange) || metadata.Monikers.Length > 0)
             {
                 // Moniker range not defined in docfx.yml/docfx.json,
                 // user should not define it in file metadata
-                if (configMonikers.Length == 0)
+                if (configedMonikers.Length == 0)
                 {
-                    return (Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange), configMonikers);
+                    errors.Add(Errors.Versioning.MonikerRangeUndefined(metadata.MonikerRange));
+                    return (errors, configedMonikers);
                 }
 
                 var fileMonikers = _rangeParser.Parse(metadata.MonikerRange);
-                var intersection = configMonikers.Intersect(fileMonikers).ToArray();
+
+                // monikerRagen takes precedence over monikers since it is more likely from user configuration
```

Comment:
I recall we agreed `monikers` should take precedence since it is more precise.

Reply:
I would prefer `MonikerRange` though as reasoned in description

Reply:
@928PJY @herohua for more inputs

Reply:
As I understand, `MonikerRang` and `Monikers` should be equal as long as the moniker definition is correct, why is `monikers` more precise here?
As mentioned in the description, the conceptual users are more likely to use `MonikerRange`, and `Monikers` are only used within combined-versioning, which is from plugin calculation.
@928PJY @herohua